Forge/src/main/java/me/ferdz/placeableitems/block/component/impl/FragileBlockComponent.java

```diff
@@ -0,0 +1,99 @@
+package me.ferdz.placeableitems.block.component.impl;
+
+import java.util.function.Supplier;
+
+import me.ferdz.placeableitems.block.component.AbstractBlockComponent;
+import me.ferdz.placeableitems.wiki.WikiBlockComponentDefinition;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.world.World;
+
+@WikiBlockComponentDefinition(description = "Right clicking this item will break it and drop an item or entity")
```

Comment:
Since this is generic, this definition holds no informational value. The downside to having these components reusable is that they will now need to be described in the block's `@WikiItem`

Reply:
Agreed and this came to mind as well when writing it. I'm not sure the best approach here.

Reply:
Yes, that's correct